app/src/androidTest/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtilsTest.kt

```diff
@@ -363,4 +541,81 @@ class FirebaseDatabaseUtilsTest {
+                    0.8,
+                ),
+                votes = 2,
+                date = LocalDateTime.now().minusDays(2),
+                usersVotes = mutableMapOf("user1" to 1, "user2" to 1),
+            ),
+            TournamentPost(
+                "user2",
+                run = Run(
+                    Path(listOf(listOf(LatLng(0.1, 1.0), LatLng(1.1, 0.2)), listOf(LatLng(9.9, 10.11)))),
+                    30,
+                    20,
+                    50,
+                ),
+                votes = 0,
+                date = LocalDateTime.now().minusDays(3),
+                usersVotes = mutableMapOf("user1" to -1, "user2" to 1),
+            ),
+        )
+        val tournament = Tournament(
+            id = "testId",
+            name = "testName",
+            description = "testDescription",
+            creatorId = "testCreatorId",
+            startDate = LocalDateTime.now(),
+            endDate = LocalDateTime.now().plusWeeks(1),
+            participants = listOf("user1", "user2"),
+            posts = posts,
+            visibility = Tournament.Visibility.PUBLIC,
+        )
+
+        val tournamentDataSnapshot = mockTournament(tournament)
+        val reformedTournament = FirebaseDatabaseUtils.mapToTournament(tournamentDataSnapshot)
+
+        assertThat(reformedTournament!!.id, `is`(tournament.id))
+        assertThat(reformedTournament.name, `is`(tournament.name))
+        assertThat(reformedTournament.description, `is`(tournament.description))
+        assertThat(reformedTournament.creatorId, `is`(tournament.creatorId))
+        assertThat(reformedTournament.startDate, `is`(tournament.startDate))
+        assertThat(reformedTournament.endDate, `is`(tournament.endDate))
+        assertThat(reformedTournament.participants, `is`(tournament.participants))
+        assertThat(reformedTournament.posts.size, `is`(tournament.posts.size))
+        assertThat(reformedTournament.visibility, `is`(tournament.visibility))
+    }
+
+    @Test
+    fun mapToTournamentReturnsNullWithNullValues() {
+        // With only null values
+        assertEquals(FirebaseDatabaseUtils.mapToTournament(null), null)
+        // With existing visibility but other values null, to test other path
+        val mockVisibility = mockSnapshot("PUBLIC")
+        val nonNullVisibilitySnapshot = mock(DataSnapshot::class.java)
+        `when`(nonNullVisibilitySnapshot.child("visibility"))
+            .thenReturn(mockVisibility)
+        assertEquals(FirebaseDatabaseUtils.mapToTournament(nonNullVisibilitySnapshot), null)
+    }
+
+    @Test
+    fun mapToTournamentReturnsNullWithNonExistingVisibilityValue() {
+        val mockVisibility = mockSnapshot("SomeVisibility")
+        val nonNullVisibilitySnapshot = mock(DataSnapshot::class.java)
+        `when`(nonNullVisibilitySnapshot.child("visibility"))
+            .thenReturn(mockVisibility)
+        assertEquals(FirebaseDatabaseUtils.mapToTournament(nonNullVisibilitySnapshot), null)
+    }
```

Comment:
Suggestion: Same here

Reply:
I can add comments in `mapToTournamentReturnsNullWithNonExistingVisibilityValue`, good suggestion! I think comments in the other tests would be really redundant though.